src/imitation/algorithms/adversarial.py

```diff
@@ -87,6 +88,19 @@ def __init__(self,
           This disables the reward wrapping that would normally replace
           the environment reward with the learned reward. This is useful for
           sanity checking that the policy training is functional.
+        error_on_unexpected_policy_change: If True, then enable training
```

Comment:
Not sure if we actually want these errors to be optional. Leaning towards nonoptional errors because the silent resets can be so tricky to detect.

In the future if there's actually a use case for this sort of modification we can maybe add this back in.

Reply:
I'm OK making these errors mandatory (i.e. removing the keyword argument and just always having the check). In this case should move some of the docstring into a comment near the `assert` statement since this is useful context to have.